grizzly/adapter/adapter.py

```diff
@@ -43,27 +47,27 @@ class Adapter(metaclass=ABCMeta):
 
     __slots__ = ("_harness", "fuzz", "monitor", "name", "remaining")
 
-    def __init__(self, name):
+    def __init__(self, name: str) -> None:
         assert isinstance(name, str)
         if not name:
             raise AdapterError("name must not be empty")
         if len(name.split()) != 1 or name.strip() != name:
             raise AdapterError("name must not contain whitespace")
-        self._harness = None
-        self.fuzz = {}
-        self.monitor = None
+        self._harness: Optional[bytes] = None
+        self.fuzz: Dict[str, Any] = {}
+        self.monitor: Optional[TargetMonitor] = None
         self.name = name
-        self.remaining = None
+        self.remaining: Optional[int] = None
 
-    def __enter__(self):
+    def __enter__(self) -> "Adapter":
         return self
 
-    def __exit__(self, *exc):
+    def __exit__(self, *exc: Any) -> None:
         self.cleanup()
 
-    def cleanup(self):
+    def cleanup(self) -> None:
```

Comment:
This (and `enable_harness`, `get_harness`) could be annotated with `typing.final`.